File: NaviTest/NaviTest/ViewController.m
//
//  ViewController.m
//  NaviTest
//
//  Created by jiazhiyi on 2019/3/15.
//  Copyright © 2019 jiazhiyi. All rights reserved.
//

#import "ViewController.h"
#import "FirstViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.view.backgroundColor = [UIColor yellowColor];
    UIButton *button = [UIButton buttonWithType:UIButtonTypeCustom];
    button.frame = CGRectMake(0, 100, 100, 100);
    button.backgroundColor = [UIColor blackColor];
    [button addTarget:self action:@selector(respondsToButton) forControlEvents:UIControlEventTouchUpInside];
    [self.view addSubview:button];
    

    self.pinkieItem.title = @"你哈";
//    self.navigationTitle = @"你哈";
}

- (void)respondsToButton
{
    FirstViewController *firstVC = [[FirstViewController alloc] init];
    [self.navigationController pushViewController:firstVC animated:YES];
}

@end

File: CustomNavigationBar/CustomNavigationBar/ViewController.swift
//
//  ViewController.swift
//  CustomNavigationBar
//
//  Created by jiazhiyi on 2019/4/4.
//  Copyright © 2019 jiazhiyi. All rights reserved.
//

import UIKit

class ViewController: PinkieViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        
        self.pinkieItem.title = "你好"
        
        self.view.backgroundColor = UIColor.white
        
        let button = UIButton(type: .custom)
        button.frame = CGRect(x: 0, y: 100, width: 100, height: 100)
        button.backgroundColor = UIColor.black
        button.addTarget(self, action: #selector(respondsToButton), for: .touchUpInside)
        self.view.addSubview(button)
    }

    @objc func respondsToButton() {
        
        let firstVC = FirstViewController()
        self.navigationController?.pushViewController(firstVC, animated: true)
    }

}

